docs/product/components/modals.html

```diff
@@ -3,31 +3,122 @@
+        <table class="wmn4 s-table s-table__bx-simple">
+            <thead>
+                <tr>
+                    <th class="s-table--cell5" scope="col">Event</th>
+                    <th class="s-table--cell2" scope="col">Element</th>
+                    <th scope="col">Description</th>
+                </tr>
+            </thead>
+            <tbody class="fs-caption">
+                <tr>
+                    <th scope="row"><code class="stacks-code">s-modal:show</code></th>
+                    <td>Controller element</td>
+                    <td>Fires immediately before showing the modal</td>
+                </tr>
+                <tr>
+                    <th scope="row"><code class="stacks-code">s-modal:shown</code></th>
+                    <td>Controller element</td>
+                    <td>Fires immediately after showing the modal</td>
+                </tr>
+                <tr>
+                    <th scope="row"><code class="stacks-code">s-modal:hide</code></th>
+                    <td>Controller element</td>
+                    <td>Fires immediately before hiding the modal</td>
+                </tr>
+                <tr>
+                    <th scope="row"><code class="stacks-code">s-modal:hidden</code></th>
+                    <td>Controller element</td>
+                    <td>Fires immediately after hiding the modal</td>
+                </tr>
+            </tbody>
+        </table>
+    </div>
+    {% header h3 | Helpers %}
+    <div class="overflow-x-auto mb32">
+        <table class="wmn4 s-table s-table__bx-simple">
+            <thead>
+                <tr>
+                    <th class="s-table--cell5" scope="col">Function</th>
+                    <th class="s-table--cell2" scope="col">Parameters</th>
+                    <th scope="col">Description</th>
+                </tr>
+            </thead>
+            <tbody class="fs-caption">
+                <tr>
+                    <th scope="row"><code class="stacks-code">Stacks.toggleModal</code></th>
+                    <td>
+                        <code class="stacks-code">element</code>: the element the <code class="stacks-code">data-controller="s-modal"</code> attribute is on <br><br>
+                        <code class="stacks-code">show</code>: whether to force show/hide the modal; toggles the modal if left undefined
+                    </td>
+                    <td>Helper to manually show an s-modal element via external JS</td>
+                </tr>
+            </tbody>
+        </table>
+    </div>
+</section>
+
 <!-- Example base modal content -->
-<aside class="s-modal js-modal-overlay js-modal-close" id="modal-base" tabindex="-1" role="dialog" aria-labelledby="modal-title" aria-describedby="modal-description" aria-hidden="true">
+<aside class="s-modal js-modal-overlay js-modal-close" id="modal-base" tabindex="-1" role="dialog" aria-labelledby="modal-title" aria-describedby="modal-description" aria-hidden="true"
```

Comment:
Will there ever be a scenario where `data-controller="s-modal"` and `data-target="modal"` won't be the same element?  It probably makes sense to ditch `this.modalTarget` and just use `this.element` everywhere.

Reply:
I don't have an example in the docs, but you could wrap a shared parent element so you can make a button toggle your modal, similar to how we do popovers (and how the other modal PR worked)